internal/api/handlers.go

```diff
@@ -154,3 +155,44 @@ func writeJSON(w http.ResponseWriter, status int, v interface{}) {
 func writeError(w http.ResponseWriter, status int, message string) {
 	writeJSON(w, status, models.Error{Error: message})
 }
+
+func (h *Handler) GetApiV1ArtifactsImage(w http.ResponseWriter, r *http.Request) {
+
+	image := r.URL.Query().Get("uri")
+	if image == "" {
+		writeError(w, http.StatusBadRequest, "Missing image URI")
+		return
+	}
+	username, password, ok := r.BasicAuth()
+	if !ok && (username != "" || password != "") {
+		writeError(w, http.StatusBadRequest, "Invalid Authorization header")
+		return
+	}
+
+	ctx := r.Context()
+	response, err := h.artifactService.GetImageMetadata(ctx, image, username, password)
+
+	if err != nil {
+		errMsg := strings.ToLower(err.Error())
+		switch {
+		case strings.Contains(errMsg, "Image not found"):
+			writeError(w, http.StatusNotFound, "not found")
+		case strings.Contains(errMsg, "authentication failed"):
+			writeError(w, http.StatusUnauthorized, "Authentication failed")
```

Comment:
It may be beneficial to create a dedicated errors package for all of these errors it will improve readability and usability